src/request.ts

```diff
@@ -77,9 +79,15 @@ const attachErrorDetails = (err: unknown) => {
 }
 
 export const addVersionToUrl = (url: string, apiVersioning: boolean, version: Version = "v3"): string => {
-  if (!apiVersioning) return url
-  if (url.includes("identity")) return url
-  return /\/v.+/g.test(url) ? url : `${url}/${version}`
+  if (!apiVersioning || url.includes("identity") || /\/v.+/g.test(url)) return url
+  const urlWithVersion = R.pipe(
+    R.split("/"), // split url [ "https:", "", "test.com", "path", "path2" ]
+    R.remove(1, 1), // Remove empty string from // in protocol
+    R.adjust(0, flippedConcat("/")), // Add back / to protocol
+    R.insert(2, String(version)), // insert and stringify version after domain
+    R.join("/"), // join url back together with slash
```

Comment:
Does it make any difference doing this vs `R.split("/"), R.insert(3, String(version)), R.join("/")`?

Reply:
God i must have just been in a ramda haze and did not even consider the empty string would just work again when joining! Thanks Luke :sweat_smile: